public/main/group/group_creation.php

```diff
@@ -203,13 +218,29 @@ function copy_value(key) {
             $group_el = [];
             $group_el[] = $form->createElement('text', 'group_'.$group_number.'_name');
             if ($allowGroupCategories) {
-                $group_el[] = $form->createElement(
-                    'select',
-                    'group_'.$group_number.'_category',
-                    null,
-                    $categories,
-                    ['id' => 'category_'.$group_number]
-                );
+                if(isset($_GET['category_id'])) {
+                    $group_el[] = $form->createElement(
+                        'select',
+                        'group_' . $group_number . '_category',
+                        null,
+                        $categories,
+                        [
+                            'id' => 'category_' . $group_number,
+                            'disabled' => 'true',
+                            'style' => 'background-color: #f0f0f0;'
+                        ]
+                    );
+                    $group_el[] = $form->createElement('hidden', 'group_'.$group_number.'_category', $_GET['category_id']);
+                    $defaults['group_'.$group_number.'_category'] = $_GET['category_id'];
```

Comment:
`$_GET['category_id']` should be filtered (because we know it's an integer) with `(int) $_GET['category_id']`

When a `$categoryId` comes from an integer field in the database, you know it's an integer so you don't have to filter it, but anything from $_GET, $_POST, $_REQUEST, even $_SESSION and other superglobals *has* to be filtered before inserting into the database or an URL or anything on a web page.